sdk/storage/AzureStorageBlob/Source/Options/ClientMethodOptions.swift

```diff
@@ -419,3 +423,11 @@ public struct UploadBlobOptions: AzureOptions, Codable, Equatable {
             encoding, timeoutInSeconds
     }
 }
+
+public typealias SnapshotDate = Date
+extension SnapshotDate: QueryStringConvertible {
```

Comment:
Not convinced this alias+extension is optimal, see below.

Reply:
Hmm.. yes, this won't work because:
```swift
typealias CatDate = Date
extension CatDate: QueryStringConvertible {
...
}

typealias DogDate = Date
extension DogDate: QueryStringConvertible {
...
}
```

This raises an invalid redundant conformance of Date to QueryStringConvertible.

Reply:
Unfortunately, `structs` can't be extended, so I can't do this either:
```swift
struct SnapshotDate: Date, CustomStringConvertible {
  let format = "blah"
  ...
}
```

Reply:
This solution has been replaced in the latest commit.